### ISSUE TITLE
Improve Jupyter representation of File instances

### ISSUE THREAD

**Issue by samsammurphy** [ecmwf/earthkit-data#48]
In Jupyter if I try to inspect a File instance by typing it as the last thing in a cell it is displayed as an empty table

![Screenshot 2023-05-04 at 09 22 38](https://user-images.githubusercontent.com/16657983/236149555-61df68d9-01ed-43e7-a7cf-483c08c49197.png)

This is because Jupyter is calling `_repr_html_()` method which is defined in the Source base class [here](https://github.com/ecmwf/earthkit-data/blob/497d60544799c88a03501754fe24f1b07fd57b57/earthkit/data/sources/__init__.py#L86) and returning this [table](https://github.com/ecmwf/earthkit-data/blob/497d60544799c88a03501754fe24f1b07fd57b57/earthkit/data/utils/html.py#L36).

Conversely, printing the file uses `__str__()` (if available, which it is) and generates this more informative output

![Screenshot 2023-05-04 at 09 46 00](https://user-images.githubusercontent.com/16657983/236155417-c2e61946-21a0-48b7-a13b-dc49433517f1.png)

I am not sure how often (in practice) the metadata in the current default table will be available (i.e. home page, licence, citation and documentation) when loading a file from source.

**Comment by samsammurphy** [comment]
Tbh I am not sure what is the best way forward. @sandorkertesz , @JamesVarndell  what do you think? One idea might be to return `__str__()` on calls to `_repr_html()`

![Screenshot 2023-05-04 at 10 09 14](https://user-images.githubusercontent.com/16657983/236160512-f53c256a-6b62-43c9-bd2b-4bc9054cb663.png)

**Comment by tlmquintino** [comment]
@samsammurphy thanks for opening this issue. This is definitely missing functionality, however I think calling it a file is not correct. We are not representing a file. The methods in the class don't relate to a file.
This is a Data abstraction. We are not abstracting the file itself, we are abstracting the data set in the file. 
So examples, should not show...
`f = earthkit.data.from_source(...)`
but
`dataset = earthkit.data.from_source(...)`
`ds = ...` ok too.

The class underneath should probably be called FileDataSet or something similar, and the output of that Sam is looking for should be in the lines of:
`FileDataSet(test6.grib,GRIBReader)`

**Comment by JamesVarndell** [comment]
I agree with @tlmquintino about changing the class name, but I also think there is scope for a better pretty HTML representation of the data for users to see when in Jupyter - think what xarray and pandas do (and indeed what individual fields already do in earthkit). Needs some more discussion before we implement anything - I'll update this issue once we have a plan, although also very happy to hear any suggestions you may have @samsammurphy!

**Comment by samsammurphy** [comment]
Hey @JamesVarndell 

Short answer = lets talk about it next week (since I'm on site)

Abstract answer = We should capture the "essence" of the data in as concise and meaningful a way as possible from the perspective of the target user(s). Ideally with userbility tests

Long answer = I agree that a richer HTML based representation is the direction we should head in. Something like xarray, but better ;)  I really like the tab style of html output from dump() method on a field. The first tab could have the "essence" of the data, the other tabs could hold the details. Most of the data I have seen in earthkit so far seems to be rasters (i.e. georeferenced, labelled, arrays) for this kind of data the essence might be: names, shape and dtype of arrays (we might even move geographic information to a separate tab, i.e. coord labels, crs, spatial resolution, etc..). We will (I believe) also support vectors (i.e. georefenced shapes). For this type of data the essence might just be a rendering of the shape (i.e. the way a shapely objects behaves in a notebook) and we could put any text information in separate tabs. This tab based approach could help organize / simplify the complexity we will need to handle (e.g. @sandorkertesz consideration of all other objects important too).

**Comment by Oisin-M** [comment]
The new high-level data objects have updated `__repr__` and `_repr_html_` methods implemented so I am therefore closing this issue
